node_modules/oae-zencoder/lib/util.js

```diff
@@ -0,0 +1,129 @@
+var TenantsAPI = require('oae-tenants/lib/api');
+var ContentConfig = require('oae-config').config('oae-content');
+
+var ZencoderConfig = require('oae-config').config('oae-zencoder');
+
+var config = null;
+
+/**
+ * Set the default config. This can be overridden for individual tenants
+ *
+ * @return {Object} Zencoder configuration
+ */
+var init = module.exports.init = function(_config) {
+    config = _config;
+};
+
+/**
+ * Get the Zencoder API values that have been configured in the Admin UI.
+ *
+ * @return {Object} The Zencoder configuration for the tenant or the generic config for the host
+ */
+var getConfig = module.exports.getConfig = function(tenantAlias) {
+    var apiKey = ZencoderConfig.getValue(tenantAlias, 'zencoder', 'apiKey');
+
+    if (apiKey) {
+        return _.extend({}, config, {
+            'apiKey': apiKey,
+            'watermark': null,
+            'amazons3-region': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-region') || config['amazons3-region'],
+            'amazons3-bucket': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-bucket') || config['amazons3-bucket'],
+            'amazons3-access-key': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-access-key') || config['amazons3-access-key'],
+            'amazons3-secret-key': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-secret-key') || config['amazons3-secret-key']
+        });
+    } else {
+        return config;
+    }
+};
+
+/**
+ * Create an encoding job in Zencoder.
+ *
+ * @param  {Object}     config              The default configuration
+ * @param  {String}     tenantAlias         The alias of the tenant whose configuration to use
+ * @param  {String}     path                The signed path to video/audio file
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Error}      callback.err        An error that occurred, if any
+ * @param  {Response}   callback.response   The HTTP Response object
+ * @param  {String}     callback.body       The body of the response
+ */
+var createJob = module.exports.createJob = function(tenantAlias, path, contentId, revisionId, mime, callback) {
+    var config = getConfig(tenantAlias);
+    var isVideo = PreviewUtil.isVideo(mime);
+
+    var tenant = TenantsAPI.getTenant(tenantAlias);
+    var baseUrl = util.format('s3://s3-%s.amazonaws.com/%s/%s', config['amazons3-region'], config['amazons3-bucket'], encodeURIComponent(contentId));
+    var extension = (isVideo) ? 'mp4' : 'mp3';
+    var filename = util.format('%s.%s', revisionId.replace(/:/g, '-'), extension);
+
+    // @See https://app.zencoder.com/docs/api/encoding for details of Zencoder's encoding settings
+    var options = {
+        'input': util.format('https://%s%s', tenant.host, path),
+        'pass_through': util.format('%s/%s', contentId, filename), // We store the S3 bucket filepath so it can later be used to create a signed URL
```

Comment:
`pass_through` is what the filepath ends up being on S3, with contentId as the 'folder' and revisionId plus the extension as the filename. We need a value for these in util.js, since that's what we're using to tell Zencoder where to put the file. In api.js we could get the filepath from the url the file ended up on by getting the substring after the third /, or we could recreate a string with the contentId, revisionId with : replaced with -, and an extension depending on `notification.output.format`, or we could use `pass_through`. I don't see a benefit to the first two options.